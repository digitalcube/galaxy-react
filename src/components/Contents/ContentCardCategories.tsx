import React, { FC, Fragment } from 'react';
import { ContentCardProps } from './ContentCard';

export type ContentCardCategoriesProps = Pick<ContentCardProps, 'categories'>
export const ContentCardCategories: FC<ContentCardCategoriesProps> = ({categories}) => {
    if (!categories) return null;
  const categoryLength = categories.length
  return (
    <div className="mb-2 guide-list-genre">
      {categories.map((category, index) => {
        const separater = index + 1 < categoryLength ? ' / ' : null
        if (!category.link) {
          return <Fragment key={index}>{category.name}{separater}</Fragment>
        }
        return (
          <Fragment key={index}>
            <a href={category.link}>{category.name}</a>
            {separater}
          </Fragment>
        )
      })}
    </div>
  )
}