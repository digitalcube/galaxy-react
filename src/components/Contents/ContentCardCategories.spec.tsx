import { render } from '@testing-library/react';
import React from 'react';
import { Default as ContentCardCategories } from './ContentCardCategories.stories';

describe('ContentCardCategories', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(
      <ContentCardCategories {...(ContentCardCategories.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
