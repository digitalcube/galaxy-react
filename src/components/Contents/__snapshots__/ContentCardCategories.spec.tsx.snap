// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCardCategories to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-2 guide-list-genre"
  >
    <a
      href="https://example.com"
    >
      domain
    </a>
     / test
  </div>
</DocumentFragment>
`;
